pyccel/parser/syntax/headers.py

```diff
@@ -150,14 +152,18 @@ def __init__(self, **kwargs):
         dtype: list fo str
         """
         self.dtype = kwargs.pop('dtype')
+        self.const = kwargs.pop('const')
 
         super(UnionTypeStmt, self).__init__(**kwargs)
 
     @property
     def expr(self):
         l = []
-        for i in self.dtype:
+        for num, i in enumerate(self.dtype):
             l += [i.expr]
+            if self.const is not None:
+                l[num]["is_const"] = True
```

Comment:
Doesn't `l` contain lists not dictionaries? I'm not sure how this works

Reply:
it contain dictionaries

Reply:
Then can you explain the line above it:
`l += [i.expr]`?

Reply:
im not sure does this line do, from what i understand all the datatypes from the header

Reply:
I think half of your sentence is missing ^^'. I didn't think you could add a list to a dictionary (hence my question). Am I wrong? I know that we can declare a dictionary from a list as follows:
```
d = dict([('a', 1), ('b', 2), ('c', 3)])
```
Does adding a list to a dictionary work in the same way?

Reply:
`i.expr` is a dictionary, `[i.expr]` is a list, and `l += [i.expr]` extends the list `l`. But this all property is written in a strange way, it's not very clear... What about this instead?
```python
l = []
for i in self.dtype:
    if self.const:
        i.expr["is_const"] = True
    l.append(i.expr)
```
or this
```python
l = [i.expr for i in self.dtype]
if self.const:
    for e in l:
        e["is_const"] = True
```

Reply:
Yes the `i.expr` is a dictionary and `l += [i.expr]` extends the list `l`